pkg/storage-init/storage-init.sh

```diff
@@ -104,16 +104,23 @@ if P3=$(findfs PARTLABEL=P3) && [ -n "$P3" ]; then
     #For systems with ext3 filesystem, try not to change to ext4, since it will brick
     #the device when falling back to old images expecting P3 to be ext3. Migrate to ext4
     #when we do usb install, this way the transition is more controlled.
-    #Any fsck error (ext3 or ext4), will lead to formatting P3 with ext4
+    #Any fsck or mount error (ext3 or ext4), will lead to formatting P3 with ext4
+    if [ "$P3_FS_TYPE" = "ext3" ]; then
```

Comment:
Right, but the following complex if statement is expected to take care of this (sand blkid issues that we need to understand better).

Reply:
@rvs If the fsck fails we reformat. But if the ext3 mount fails (which is what happened to three test devices) we do not reformat in the current code.

Note that I don't know exactly how we got to that state because I was severly confused by the zedbox blkid (which is what is in the pillar container) and the real blkid (which is what is in the hostfs and the storage-init container) work differently.
I think I have one device which has not yet been repaired (but it isn't reachable) in this state so I'll see if I can unconfuse myself. But the logs we have clearly indicate that the devices tried mount -t ext3 when the filesystem was already ext4, which indicates that blkid in storage-init returned the wrong thing. Those devices were running 5.6.1 but I couldn't see any changed to which packages we use in the storage-init Dockerfile which would explain why 5.6.1 would work differently

Reply:
@rvs I've verified that on one more device (running 5.4.1 in this case; others were running 5.6.1) the blkid is saying "ext3" but lsblk -fs -o FSTYPE is saying ext4. That vintage system only have the busybox blkid.

Note that on current master (don't know since when) the blkid in the storage-init container is NOT the busybox one, hence it should work. But still don't understand why 5.4.1/5.6.1 worked on most of the devices. There must been some leftover state in some partition table which is confusing the busybox blkid.